### PR TITLE
[workspace] Add integration test cases for workspace assets.

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description

Add integration test cases for workspace assets.

### Issues Resolved

[List any issues this PR will resolve]

### Check List

- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+          (value) => (targetWorkspaceId = value)
+        );
+      }
+    });
+
+    after(() => {
+      cy.deleteWorkspaceByName(sourceWorkspaceName);
+      cy.deleteWorkspaceByName(targetWorkspaceName);
+      sourceWorkspaceId = '';
+      targetWorkspaceId = '';
```
Comment (Hailong-am) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js:60]: shall we uninstall sample data when do test data clean up?

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        headers: {
+          'Osd-Xsrf': 'osd-fetch',
+        },
+      })
+        .then((resp) => {
+          cy.wrap(resp.body.saved_objects).should('have.length', 1);
+          cy.wrap(resp.body.saved_objects[0].id);
+        })
+        .then((indexPatternId) => {
+          cy.visit(`${BASE_PATH}/app/indexPatterns/patterns/${indexPatternId}`);
```
Comment (Hailong-am) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js:130]: we will block index pattern access outside of workspace in future right? can we add a comment here, after this link been blocked, we could remove this this test?

**Review by Hailong-am — APPROVED** [review]

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js (hunk 1 above)
Comment (SuZhou-Joe) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js:60]: Yeah, we should uninstall to remove the data index.

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js (hunk 2 above)
Comment (SuZhou-Joe) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js:141]: Yes, good catch. Comment added.

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+if (Cypress.env('WORKSPACE_ENABLED')) {
+  describe('Workspace assets', () => {
+    before(() => {
+      cy.deleteWorkspaceByName(sourceWorkspaceName);
+      cy.deleteWorkspaceByName(targetWorkspaceName);
+      if (Cypress.env('DATASOURCE_MANAGEMENT_ENABLED')) {
+        cy.createDataSourceNoAuth().then((result) => {
+          datasourceId = result[0];
+          expect(datasourceId).to.be.a('string').that.is.not.empty;
```
Comment (raintygao) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js:38]: Is this used to verify the data source creation successfully?

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        setupWorkspace(sourceWorkspaceName).then(
+          (value) => (sourceWorkspaceId = value)
+        );
+        setupWorkspace(targetWorkspaceName).then(
+          (value) => (targetWorkspaceId = value)
+        );
+      }
+    });
+
+    after(() => {
```
Comment (raintygao) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js:56]: Do we need to call `cy.deleteDataSource(id)` in the after hook?

Comment (SuZhou-Joe) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js:56]: Yeah, we should.

**Review by raintygao — APPROVED** [review]

File: cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+    // TODO: this test case should be removed once index pattern access outside of workspace being blocked in the future.
+    it('Should not show set as default button when outside workspace', () => {
+      cy.request({
+        url: `${BASE_PATH}/api/opensearch-dashboards/management/saved_objects/_find?workspaces=${targetWorkspaceId}&page=1&type=index-pattern`,
+        headers: {
+          'Osd-Xsrf': 'osd-fetch',
+        },
+      })
+        .then((resp) => {
+          cy.wrap(resp.body.saved_objects).should('have.length', 1);
```
Comment (ruanyl) [inline cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js:138]: Do you need `cy.wrap` here?

Comment (SuZhou-Joe) [reply cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js:138]: wrap is for making it chainable with an assertion actually.

**Pull request merged**